### PR TITLE
Manual Balance Plugin - brightness contrast adjustments

### PR DESCRIPTION
**Pull request by andenixa**
- Added brightness and contrast adjustment to manual color plugin
image

### CODE REVIEW

File: plugins/convert/color/manual_balance_defaults.py [hunk 1]
```diff
@@ -114,4 +114,24 @@
         "gui_radio": False,
         "fixed": True,
     },
+    "constrast": {
```
Comment (torzdf) [inline plugins/convert/color/manual_balance_defaults.py]: contrast

File: plugins/convert/color/manual_balance.py [hunk 2]
```diff
@@ -21,6 +21,21 @@ def process(self, old_face, new_face, raw_mask):
                 image[:, :, idx] = image[:, :, idx] * (1 + adjustment[idx])
 
         image = self.convert_colorspace(image * 255.0, to_bgr=True)
+
```
Comment (torzdf) [inline plugins/convert/color/manual_balance.py]: `image = self.adjust_contrast(image)`

File: plugins/convert/color/manual_balance.py [hunk 3]
```diff
@@ -21,6 +21,21 @@ def process(self, old_face, new_face, raw_mask):
...
+        constrast = max(-126, int(round(self.config["constrast"] * 1.27)))
+        brightness = max(-126, int(round(self.config["brightness"] * 1.27)))
+
+        if constrast or brightness:
+            image = self.adjustContrast(image, constrast, brightness)
+        return image
+
+    def adjustContrast(self, image, contrast, brightness):
```
Comment (torzdf) [inline plugins/convert/color/manual_balance.py]: snake case for functions (`adjust_contrast`)
Remove contrast, brightness from function call

File: plugins/convert/color/manual_balance.py [hunk 4]
```diff
@@ -21,6 +21,21 @@ def process(self, old_face, new_face, raw_mask):
                 image[:, :, idx] = image[:, :, idx] * (1 + adjustment[idx])
 
         image = self.convert_colorspace(image * 255.0, to_bgr=True)
+
+        constrast = max(-126, int(round(self.config["constrast"] * 1.27)))
+        brightness = max(-126, int(round(self.config["brightness"] * 1.27)))
+
+        if constrast or brightness:
+            image = self.adjustContrast(image, constrast, brightness)
```
Comment (torzdf) [inline plugins/convert/color/manual_balance.py]: remove 25-29

File: plugins/convert/color/manual_balance.py [hunk 5]
```diff
@@ -21,6 +21,21 @@ def process(self, old_face, new_face, raw_mask):
...
+        return image
+
+    def adjustContrast(self, image, contrast, brightness):
+        """
+        Adjust image contrast and brightness.
+        """
```
Comment (torzdf) [inline plugins/convert/color/manual_balance.py]:
```py
        contrast = max(-126, int(round(self.config["contrast"] * 1.27)))
        brightness = max(-126, int(round(self.config["brightness"] * 1.27)))
        if not contrast and not brightness:
            return image
```

Comment (andenixa) [reply plugins/convert/color/manual_balance.py]: my version is more likely to happen

Comment (andenixa) [reply plugins/convert/color/manual_balance.py]: done

File: plugins/convert/color/manual_balance.py [hunk 6]
```diff
@@ -21,6 +21,22 @@ def process(self, old_face, new_face, raw_mask):
...
         image = self.convert_colorspace(image * 255.0, to_bgr=True)
+        image = self.adjust_contrast(image)
+        return image
+
+    def adjust_contrast(self, image):
+        """
+        Adjust image contrast and brightness.
+        """
+        contrast  = max(-126, int(round(self.config["contrast"] * 1.27)))
+        brightness = max(-126, int(round(self.config["brightness"] * 1.27)))
```
[the remark this answers is not in the document]
Comment (andenixa) [reply plugins/convert/color/manual_balance.py]: hooray for amazing nitpicking 
`round` doesn't return an `int` btw

File: plugins/convert/color/manual_balance.py [hunk 7]
```diff
@@ -21,6 +21,22 @@ def process(self, old_face, new_face, raw_mask):
...
+        
+        if not contrast or not brightness:
+            return image
+        
+        image = np.rint(image * 255.0).astype("uint8")
+        image = np.clip(image * (contrast/127+1) - contrast + brightness, 0, 255)
+        image = np.clip(np.divide(image, 255, dtype=np.float32), .0, 1.0)
```
[the remark this answers is not in the document]
Comment (andenixa) [reply plugins/convert/color/manual_balance.py]: I am aware of that, thank you

File: plugins/convert/color/manual_balance.py [hunk 8]
```diff
@@ -21,6 +21,22 @@ def process(self, old_face, new_face, raw_mask):
...
+        Adjust image contrast and brightness.
+        """
+        contrast  = max(-126, int(round(self.config["contrast"] * 1.27)))
+        brightness = max(-126, int(round(self.config["brightness"] * 1.27)))
+        
+        if not contrast or not brightness:
+            return image
+        
+        image = np.rint(image * 255.0).astype("uint8")
+        image = np.clip(image * (contrast/127+1) - contrast + brightness, 0, 255)
```
[the remark this answers is not in the document]
Comment (andenixa) [reply plugins/convert/color/manual_balance.py]: yeah I guess still `python2` thinking for me

**Pull request merged**